systest build failed on centos 7.6

* Fedora 29 build is ok.
*  centos 7  test environment
```
[pi@centos7 rust-openssl]$ cat /etc/redhat-release
CentOS Linux release 7.6.1810 (Core)
[pi@centos7 rust-openssl]$
[pi@centos7 rust-openssl]$ git log -1
commit ec8aadb5183058ae68db8f2a52e17994b4770cf4
Merge: 6f9401f 1ff853a
Author: Steven Fackler <sfackler@gmail.com>
Date:   Tue Jan 22 19:01:08 2019 -0800

    Merge pull request #1043 from sfackler/circle-21

    Migrate circle config to 2.1 goodness
[pi@centos7 rust-openssl]$
[pi@centos7 rust-openssl]$ git remote -v
origin  https://github.com/sfackler/rust-openssl.git (fetch)
origin  https://github.com/sfackler/rust-openssl.git (push)
[pi@centos7 rust-openssl]$

```
* systest build error
```
[pi@centos7 systest]$ pwd
/home/pi/github/rust-openssl/systest
[pi@centos7 systest]$ cargo build
<snipped>
   Compiling syntex_pos v0.59.1
   Compiling serde_json v1.0.37
   Compiling syntex_errors v0.59.1
   Compiling syntex_syntax v0.59.1
   Compiling ctest v0.1.7
   Compiling systest v0.1.0 (/home/pi/github/rust-openssl/systest)
error: failed to run custom build command for `systest v0.1.0 (/home/pi/github/rust-openssl/systest)`
process didn't exit successfully: `/home/pi/github/rust-openssl/target/debug/build/systest-601d81dd1e87f29b/build-script-build` (exit code: 101)
--- stdout
-----------------------------------------
cargo:rerun-if-changed=../openssl-sys/src/lib.rs
cargo:rerun-if-changed=../openssl-sys/src/macros.rs
cargo:rerun-if-changed=../openssl-sys/src/aes.rs
cargo:rerun-if-changed=../openssl-sys/src/asn1.rs
cargo:rerun-if-changed=../openssl-sys/src/bio.rs
cargo:rerun-if-changed=../openssl-sys/src/bn.rs
cargo:rerun-if-changed=../openssl-sys/src/cms.rs
cargo:rerun-if-changed=../openssl-sys/src/conf.rs
cargo:rerun-if-changed=../openssl-sys/src/crypto.rs
cargo:rerun-if-changed=../openssl-sys/src/dh.rs
cargo:rerun-if-changed=../openssl-sys/src/dsa.rs
cargo:rerun-if-changed=../openssl-sys/src/dtls1.rs
cargo:rerun-if-changed=../openssl-sys/src/ec.rs
cargo:rerun-if-changed=../openssl-sys/src/err.rs
cargo:rerun-if-changed=../openssl-sys/src/evp.rs
cargo:rerun-if-changed=../openssl-sys/src/hmac.rs
cargo:rerun-if-changed=../openssl-sys/src/obj_mac.rs
cargo:rerun-if-changed=../openssl-sys/src/object.rs
cargo:rerun-if-changed=../openssl-sys/src/ocsp.rs
cargo:rerun-if-changed=../openssl-sys/src/ossl_typ.rs
cargo:rerun-if-changed=../openssl-sys/src/pem.rs
cargo:rerun-if-changed=../openssl-sys/src/pkcs7.rs
cargo:rerun-if-changed=../openssl-sys/src/pkcs12.rs
cargo:rerun-if-changed=../openssl-sys/src/rand.rs
cargo:rerun-if-changed=../openssl-sys/src/rsa.rs
cargo:rerun-if-changed=../openssl-sys/src/sha.rs
cargo:rerun-if-changed=../openssl-sys/src/srtp.rs
cargo:rerun-if-changed=../openssl-sys/src/ssl.rs
cargo:rerun-if-changed=../openssl-sys/src/ssl3.rs
cargo:rerun-if-changed=../openssl-sys/src/stack.rs
cargo:rerun-if-changed=../openssl-sys/src/tls1.rs
cargo:rerun-if-changed=../openssl-sys/src/x509.rs
cargo:rerun-if-changed=../openssl-sys/src/x509_vfy.rs
cargo:rerun-if-changed=../openssl-sys/src/x509v3.rs
OPT_LEVEL = Some("0")
HOST = Some("x86_64-unknown-linux-gnu")
CC_x86_64-unknown-linux-gnu = None
CC_x86_64_unknown_linux_gnu = None
HOST_CC = None
CC = None
CFLAGS_x86_64-unknown-linux-gnu = None
CFLAGS_x86_64_unknown_linux_gnu = None
HOST_CFLAGS = None
CFLAGS = None
DEBUG = Some("true")
running: "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-Wall" "-Wextra" "-Wall" "-Wextra" "-Werror" "-Wno-unused-parameter" "-Wno-type-limits" "-o" "/home/pi/github/rust-openssl/target/debug/build/systest-17e685b11d513877/out/all.o" "-c" "/home/pi/github/rust-openssl/target/debug/build/systest-17e685b11d513877/out/all.c"
cargo:warning=/home/pi/github/rust-openssl/target/debug/build/systest-17e685b11d513877/out/all.c: In function ‘__test_size_SRP_CTX’:
cargo:warning=/home/pi/github/rust-openssl/target/debug/build/systest-17e685b11d513877/out/all.c:9672:64: error: ‘SRP_CTX’ undeclared (first use in this function)
cargo:warning=             uint64_t __test_size_SRP_CTX(void) { return sizeof(SRP_CTX); }
cargo:warning=                                                                ^
cargo:warning=/home/pi/github/rust-openssl/target/debug/build/systest-17e685b11d513877/out/all.c:9672:64: note: each undeclared identifier is reported only once for each function it appears in
cargo:warning=/home/pi/github/rust-openssl/target/debug/build/systest-17e685b11d513877/out/all.c: In function ‘__test_align_SRP_CTX’:
cargo:warning=/home/pi/github/rust-openssl/target/debug/build/systest-17e685b11d513877/out/all.c:9673:70: error: ‘SRP_CTX’ undeclared (first use in this function)
cargo:warning=             uint64_t __test_align_SRP_CTX(void) { return __alignof__(SRP_CTX); }
cargo:warning=                                                                      ^
cargo:warning=/home/pi/github/rust-openssl/target/debug/build/systest-17e685b11d513877/out/all.c: In function ‘__test_size_SRP_CTX’:
cargo:warning=/home/pi/github/rust-openssl/target/debug/build/systest-17e685b11d513877/out/all.c:9672:13: error: control reaches end of non-void function [-Werror=return-type]
cargo:warning=             uint64_t __test_size_SRP_CTX(void) { return sizeof(SRP_CTX); }
cargo:warning=             ^
cargo:warning=cc1: all warnings being treated as errors
exit code: 1

--- stderr
thread 'main' panicked at '

Internal error occurred: Command "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-Wall" "-Wextra" "-Wall" "-Wextra" "-Werror" "-Wno-unused-parameter" "-Wno-type-limits" "-o" "/home/pi/github/rust-openssl/target/debug/build/systest-17e685b11d513877/out/all.o" "-c" "/home/pi/github/rust-openssl/target/debug/build/systest-17e685b11d513877/out/all.c" with args "cc" did not execute successfully (status code exit code: 1).

', /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/cc-1.0.28/src/lib.rs:2314:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.

[pi@centos7 systest]$

```

--- a/openssl-sys/src/ossl_typ.rs
+++ b/openssl-sys/src/ossl_typ.rs
@@ -947,6 +947,7 @@ cfg_if! {
         }
 
         #[repr(C)]
+        #[cfg(not(osslconf = "OPENSSL_NO_SRP"))]
         pub struct SRP_CTX {
             SRP_cb_arg: *mut c_void,
             TLS_ext_srp_username_callback: *mut c_void,

